wasp/src/main/java/com/orhanobut/wasp/Wasp.java

```diff
@@ -51,21 +67,36 @@ private Wasp(Builder builder) {
      */
     public static class Image {
 
+        private static ImageHandler imageHandler;
+        private static WaspImageHandler.ImageCache imageCache;
+        private static WaspImageHandler.ImageNetworkHandler imageNetworkHandler;
+
         public static WaspImage.Builder from(String path) {
             if (TextUtils.isEmpty(path)) {
                 throw new IllegalArgumentException("Path cannot be empty or null");
             }
             return new WaspImage.Builder()
-                    .setImageHandler(newImageHandler())
-                    .setLogLevel(logLevel)
+                    .setImageHandler(getImageHandler())
                     .from(path);
         }
 
-        private static ImageHandler newImageHandler() {
+        private static ImageHandler getImageHandler() {
             if (context == null) {
                 throw new NullPointerException("Wasp.Builder should be instantiated first");
             }
-            return new VolleyImageHandler(context);
+            if (imageHandler == null) {
+                imageCache = new BitmapWaspCache();
+                imageNetworkHandler = new VolleyImageNetworkHandler(context);
+                imageHandler = new WaspImageHandler(imageCache, imageNetworkHandler);
```

Comment:
We only need imageHandler here, so we can remove other fields and use like this:
<pre>
imageHandler = new WaspImageHandler(new BitmapWaspCache(), new VolleyImageNetworkHandler(context));
</pre>